### DISCUSSION TITLE
Synology NAS Docker - Could not poll the database: SQLSTATE[HY000] [2002] - I missed something

### DISCUSSION THREAD

**Discussion by aibaron** [firefly-iii/firefly-iii#8308] [Firefly III]
I'm getting the below issue:

```
Could not poll the database: SQLSTATE[HY000] [2002] Connection refused (Connection: mysql, SQL: select `id`, `name`, `data` from 
`configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1)

This error occured in file /var/www/html/app/Support/FireflyConfig.php on line 65 with code 0.
```
![image](https://github.com/firefly-iii/firefly-iii/assets/50992992/e2bee4c4-a0da-48da-acde-0339246aeb6d)

I am working on having it set up on my Synology NAS (DSM 7.2.1-69057 Update 3):
### What I did
- I followed [this guide](https://medium.com/@tzam/firefly-iii-setup-on-synology-docker-32a36bb89a35)
- Installed MariaDB and phpmyadmin
- Created the firefly database and fireflyadmin user in the phpmyadmin UI
- Downloaded the latest fireflyiii image and ran it with the settings added in the medium.com article

![image](https://github.com/firefly-iii/firefly-iii/assets/50992992/6a5724e9-f678-40ef-8ff7-37cd1fc4dc3b)
### Troubleshooting
- After receiving the above error, I tried changing the DB_HOST variable. 
  - First it was 192.168.1.xxx/phpmyadmin (.xxx is my synology NAS IP. This is how to get to the phpmyadmin UI)
  - Then I removed the phpmyadmin, changed to localhost, tried adding different ports, etc.
- I found a few other users with similar issues, but I don't think they were close enough to what I'm getting. 
  - https://github.com/orgs/firefly-iii/discussions/8133 and https://github.com/orgs/firefly-iii/discussions/7806

Any help is greatly appreciated! Thanks!!

**Comment by JC5** [comment]
It's not "localhost", that's for sure. The 192.xxx address is for the *outside* of Docker.

If you do this, what is the IP that is returned? Bet you that's the one you need:

```
docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' DATABASECONTAINERID
```

**Reply by aibaron** [reply]
Firstly, thanks for the reply!

What do you mean by the 192.168.1.165 address is for _outside_ of Docker? that is the private IP address of the Synology. 
I tried to run that, but I'm not sure which container ID to put in. You mentioned 'DATABASE' container, so I'm thinking the MariaDB; however, that's being run from the Synology package center (see screenshot below) so as far as I'm aware doesn't have a containerID. I could paste in the containerID of the firefly container, but I don't think that's what you meant.
![image](https://github.com/firefly-iii/firefly-iii/assets/50992992/fa6ced12-a55a-4ae6-af29-0f58ea7f21e0)

I saw on the general tab that it added firefly to the bridge network, so I put in that address 172.17.0.5 on a whim, but it also didn't work.
![image](https://github.com/firefly-iii/firefly-iii/assets/50992992/821c0175-4093-4c16-b0c8-a58b55f01f91)

Basically - I'm not sure how to find the IP Address of the database when the way I get to the database manager is with 192.168.1.165/phpmyadmin where 192.168.1.165 is the local ip of my synology.

**Reply by JC5** [reply]
You're close. The mysql container should also have an IP address.

**Reply by figadore** [reply]
I'm having this same problem, but only after I completely rebuilt the docker-compose stack. This time around my db container has no IP. There is no `NetworkSettings` section under `docker inspect`. Will try recreating again later to see if it's a fluke (unless there's an easier fix)
